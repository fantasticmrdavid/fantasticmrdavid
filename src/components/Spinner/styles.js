import styled, { css, keyframes } from 'styled-components';
import theme from 'styles/theme';

const load = keyframes`
  0%,
  80%,
  100% {
    box-shadow: 0 0;
    height: 4em;
  }
  40% {
    box-shadow: 0 -2em;
    height: 5em;
  }
`;

const Bar = styled.div`
  color: ${theme.colorWhite};
  animation: ${css`${load} 1s infinite ease-in-out`};
  width: 1em;
  height: 4em;
`;

export const Container = styled.div`
  position: relative;
  height: 160px;
`;

export const Spinner = styled(Bar)`
  position: relative;
  font-size: 11px;
  text-indent: -9999em;
  margin: 88px auto;
  background: ${theme.colorWhite};
  transform: translateZ(0);
  animation-delay: -0.16s;

  &:before,
  &:after {
    position: absolute;
    top: 0;
    content: '';
    background: ${theme.colorWhite};
    animation: ${css`${load} 1s infinite ease-in-out`};
    width: 1em;
    height: 4em;
  }

  &:before {
    left: -1.5em;
    animation-delay: -0.32s;
  }

  &:after {
    left: 1.5em;
  }
`;
