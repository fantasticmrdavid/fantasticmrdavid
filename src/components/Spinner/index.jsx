import React from 'react';
import * as styles from './styles';

export default () => {
  const { Spinner } = styles;
  return <Spinner />;
};
