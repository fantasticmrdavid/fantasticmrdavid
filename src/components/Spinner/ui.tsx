import React, { memo } from 'react';
import * as styles from './styles';

const { Container, Spinner } = styles;

export default memo(() => {
  return <Container><Spinner /></Container>;
});
