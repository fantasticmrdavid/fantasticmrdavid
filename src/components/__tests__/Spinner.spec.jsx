import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from 'components/Spinner';

describe('Component -> Spinner', () => {
  it('should render the Spinner without throwing an error', () => {
    expect(renderer.create(<Spinner />)).toMatchSnapshot();
  });
});
