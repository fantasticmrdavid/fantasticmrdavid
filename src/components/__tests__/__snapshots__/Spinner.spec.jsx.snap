// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Spinner should render the Spinner without throwing an error 1`] = `
<div
  className="styles__Container-sc-12l0t9s-1 dqthrM"
>
  <div
    className="styles__Bar-sc-12l0t9s-0 styles__Spinner-sc-12l0t9s-2 lbkoAF"
  />
</div>
`;
